Riot/Modules/Onboarding/AuthenticationCoordinator.swift

```diff
@@ -110,23 +110,29 @@ final class AuthenticationCoordinator: NSObject, AuthenticationCoordinatorProtoc
     
     /// Starts the authentication flow.
     @MainActor private func startAuthenticationFlow() async {
-        do {
-            let flow: AuthenticationFlow = initialScreen == .login ? .login : .register
-            let homeserverAddress = authenticationService.state.homeserver.addressFromUser ?? authenticationService.state.homeserver.address
-            try await authenticationService.startFlow(flow, for: homeserverAddress)
-        } catch {
-            MXLog.error("[AuthenticationCoordinator] start: Failed to start")
-            displayError(message: error.localizedDescription)
-            return
+        let flow: AuthenticationFlow = initialScreen == .login ? .login : .register
+        if initialScreen != .selectServerForRegistration {
+            do {
+                let homeserverAddress = authenticationService.state.homeserver.addressFromUser ?? authenticationService.state.homeserver.address
+                try await authenticationService.startFlow(flow, for: homeserverAddress)
+            } catch {
+                MXLog.error("[AuthenticationCoordinator] start: Failed to start")
+                displayError(message: error.localizedDescription)
+                return
+            }
         }
-        
-        switch initialScreen {
-        case .registration:
-            showRegistrationScreen()
-        case .selectServerForRegistration:
-            showServerSelectionScreen()
-        case .login:
-            showLoginScreen()
+
+        if authenticationService.state.homeserver.needsFallback {
+            showFallback(for: flow)
```

Comment:
I think I've made a mistake suggesting that we did this here. Now that you've added the button to the Login screen I think we should match it and do so for the registration screen too. Then the user will always have the option to pick another server which they won't in this instance. Sorry, my bad. 😬

Reply:
Updated.